components/ai_chat/browser/ai_chat_tab_helper.cc

```diff
@@ -92,6 +92,9 @@ const std::vector<ConversationTurn>& AIChatTabHelper::GetConversationHistory() {
 void AIChatTabHelper::OnConversationActiveChanged(bool is_conversation_active) {
   is_conversation_active_ = is_conversation_active;
   DVLOG(3) << "Conversation active changed: " << is_conversation_active;
+  if (MaybePopPendingRequests()) {
+    return;
+  }
   MaybeGeneratePageText();
```

Comment:
Updated `OnTabContentRetrieved` for the time being.

Reply:
That's true, we can't ask about the tab contents if you start the conversation from the url bar. I think that's ok for now to avoid the confusion of the context changing, but we can evaluate after this merges and we try it out. A new conversation could be started (or refresh the page) to have a conversation with the page context.

Reply:
Agreed. Changed to return from `MaybeGeneratePageText()` and left a comment.